File: backend/product-client/src/main/java/org/andante/subwoofers/dto/SubwoofersVariantOutputDTO.java
package org.andante.subwoofers.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.andante.product.dto.ProductVariantOutputDTO;

@SuperBuilder
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class SubwoofersVariantOutputDTO extends ProductVariantOutputDTO {

    private String color;
    private Long subwoofersId;
}

File: frontend/src/store/api/result/dto/product/variant/SubwoofersVariantOutputDTO.ts
import { ProductVariantOutputDTO } from "./ProductVariantOutputDTO";

export interface SubwoofersVariantOutputDTO extends ProductVariantOutputDTO {
    color: string,
    subwoofersId: number,
};

export function isSubwoofersVariantOutputDTO(variant: ProductVariantOutputDTO): variant is SubwoofersVariantOutputDTO {
    return "subwoofersId" in variant;
}

export function isSubwoofersVariantOutputDTOArray(variants: ProductVariantOutputDTO[]): variants is SubwoofersVariantOutputDTO[] {
    return variants.every(isSubwoofersVariantOutputDTO);
}